src/app/pages/payments/payments.component.html

```diff
@@ -0,0 +1,311 @@
+  <form class="payment-form" [formGroup]="form" (ngSubmit)="onSubmitPaymentDetails(form)">
+    <ngx-spinner></ngx-spinner>
+    <div class="heading">Pay with International Card</div>
+    <div class="container">
+      <div>
+        Card Number:
+      </div>
+      <input formControlName="cardno" class="form-control" [ngClass]="{
+          'is-invalid':
+            (form.get('cardno').touched || form.get('cardno').dirty) &&
+            !form.get('cardno').valid
+        }" />
+      <span class="invalid-feedback">
+        <span *ngIf="
+            (form.get('cardno').touched || form.get('cardno').dirty) &&
+            form.get('cardno').errors?.required
+          ">
+          Card Number is required
+          <br />
+        </span>
+        <span *ngIf="
+            (form.get('cardno').touched || form.get('cardno').dirty) &&
+            form.controls['cardno'].errors?.minlength
+          ">
+          Card number should not be less than 8 integers
+          <br />
+        </span>
+        <span *ngIf="
+            (form.get('cardno').touched || form.get('cardno').dirty) &&
+            form.get('cardno').errors?.pattern
+          ">
+          Card number should be an integer
+        </span>
+      </span>
+      <div>
+        CVV Number:
+      </div>
+      <input formControlName="cvv" class="form-control" [ngClass]="{
+          'is-invalid':
+            (form.get('cvv').touched || form.get('cvv').dirty) &&
+            !form.get('cvv').valid
+        }" />
+      <span class="invalid-feedback">
+        <span *ngIf="
+            (form.get('cvv').touched || form.get('cvv').dirty) &&
+            form.get('cvv').errors?.pattern
+          ">
+          CVV should be a number
+          <br />
+        </span>
+        <span *ngIf="
+            (form.get('cvv').touched || form.get('cvv').dirty) &&
+            form.get('cvv').errors?.required
+          ">
+          CVV is required
+          <br />
+        </span>
+        <span *ngIf="
+            (form.get('cvv').touched || form.get('cvv').dirty) &&
+            form.controls['cvv'].errors?.minlength
+          ">
+          CVV number should not be less than 3 integers
+          <br />
+        </span>
+        <span *ngIf="
+            (form.get('cvv').touched || form.get('cvv').dirty) &&
+            form.controls['cvv'].errors?.maxlength
+          ">
+          CVV number should not be more than 4 integers
+          <br />
+        </span>
+      </span>
+      <div>
+        Card Expiry:
+      </div>
+      <div>
+        <select formControlName="expirymonth" class="form-control left" placeholder="month" [ngClass]="{
+            'is-invalid':
+              (form.get('expirymonth').touched ||
+                form.get('expirymonth').dirty) &&
+              !form.get('expirymonth').valid
+          }">
+          <option value="" value="null" disabled="true" [selected]="true">Month</option>
+          <option value="01">01</option>
```

Comment:
@patrickf949 these values are hardcoded, which is unnecessary.  use the *ngFor directive and avoid repetition

Reply:
Yeah, didn't feel right doin that.
done